core/templates/components/forms/schema-based-editors/schema-based-editor.directive.ts

```diff
@@ -69,3 +69,26 @@ angular.module('oppia').directive('schemaBasedEditor', [
       controller: [function() {}]
     };
   }]);
+
+import { Directive, ElementRef, Injector, Input, Output, EventEmitter } from '@angular/core';
+import { UpgradeComponent } from '@angular/upgrade/static';
+import { Schema } from 'services/schema-default-value.service';
+
+@Directive({
+  selector: 'schema-based-editor'
+})
+export class SchemaBasedEditorDirective extends UpgradeComponent {
```

Comment:
Actually one question, should we call this a component now?

/cc @srijanreddy98

Reply:
No, it should be a directive, right? It has an `@Directive` decorator.

Reply:
Where is this been used in this PR? I don't see any test file for this directive? Also, are we expecting to remove the old schema-based-editor directive of angualrJS?

Reply:
We are not migrating schema-based-editor here, we created a wrapper on the angularjs directive using UpgradeComponent, so that, we can use angularjs directive in an angular component. schema-based-editor cannot be migrated yet as it depends on migration of ck-editor and $compile directive. So, we can only use it like this for now. It is used in `translation-modal.component.html` line 40.

Reply:
The syntax for using `UpgradeComponent` uses Directive instead of Component https://angular.io/api/upgrade/static/UpgradeComponent
The reason which I can think of is, if we use `@Component`, then we have to provide a template which may override the template of the angularjs component which we are upgrading.